Source/BSN.Commons/Infrastructure/Redis/RedisDbContext.cs

```diff
@@ -0,0 +1,43 @@
+﻿using Microsoft.Extensions.Options;
+using Redis.OM;
+using Redis.OM.Contracts;
+using Redis.OM.Searching;
+using StackExchange.Redis;
+
+namespace BSN.Commons.Infrastructure.Redis
+{
+    /// <summary>
+    /// Redis Database Context
+    /// </summary>
+    public class RedisDbContext : IRedisDbContext
+    {
+        /// <summary>
+        /// Constructor of Redis Database Context
+        /// </summary>
+        /// <param name="options">Redis Connection Options</param>
+        public RedisDbContext(IOptions<RedisConnectionOptions> options)
+        {
+            _provider = new RedisConnectionProvider(options.Value.ConnectionString);
+        }
+
+        /// <inheritdoc/>
+        public int SaveChanges()
+        {
+            throw new System.NotImplementedException("We don't have a way to save changes on redis om yet.");
```

Comment:
what is meaning of 

```
        /// <summary>
        /// Saves the current state of the collection, overriding what was initially materialized.
        /// </summary>
        void Save();
```

in https://github.com/redis/redis-om-dotnet/blob/main/src/Redis.OM/Searching/IRedisCollection.cs

Reply:
this one does save the changes on a specific collection of type T but DbContext does not have a type
and also inside the repository we get the provider not the collection

is it ok to get the collection of type T inside the repo from DbContext?

Reply:
as we talked we cant implement it yet.